testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/ClientPoliciesTest.java

```diff
@@ -1279,6 +1281,80 @@ public void testHolderOfKeyEnforceExecutor() throws Exception {
         }
     }
 
+
+    @Test
+    public void testExtendedClientPolicyIntefacesForClientRegistrationPolicyMigration() throws ClientRegistrationException, ClientPolicyException {
+        String policyName = "MyPolicy";
+        String clientName = "ByAdmin-App" + KeycloakModelUtils.generateId().substring(0, 7);
+        String executorName = "TestRaiseExeptionExecutor";
+
+        createPolicy(policyName, DefaultClientPolicyProviderFactory.PROVIDER_ID, null, null, null);
+        logger.info("... Created Policy : " + policyName);
+
+        createCondition("AnyClientConditionFactory", AnyClientConditionFactory.PROVIDER_ID, null, (ComponentRepresentation provider) -> {
+        });
+        registerCondition("AnyClientConditionFactory", policyName);
+        logger.info("... Registered Condition : AnyClientConditionFactory");
+
+        createExecutor(executorName, TestRaiseExeptionExecutorFactory.PROVIDER_ID, null, (ComponentRepresentation provider) -> {
+            provider.getConfig().put(TestRaiseExeptionExecutorFactory.TARGET_CP_EVENTS, new ArrayList<>(Arrays.asList(ClientPolicyEvent.REGISTERED.toString())));
+        });
+        registerExecutor(executorName, policyName);
+        logger.info("... Registered Executor : " + executorName);
+
+        String clientId = null;
+        try {
+            try {
+                createClientByAdmin(clientName, (ClientRepresentation clientRep) -> {
+                });
+                fail();
+            } catch (ClientPolicyException cpe) {
+                assertEquals(ClientPolicyEvent.REGISTERED.toString(), cpe.getError());
+            }
+
+            updateExecutor(executorName, (ComponentRepresentation provider) -> {
+                provider.getConfig().put(TestRaiseExeptionExecutorFactory.TARGET_CP_EVENTS, new ArrayList<>(Arrays.asList(ClientPolicyEvent.UPDATED.toString())));
+            });
+
+            clientId = getClientByAdminWithName(clientName).getId();
+            assertEquals(true, getClientByAdmin(clientId).isEnabled());
+            try {
+                updateClientByAdmin(clientId, (ClientRepresentation clientRep) -> {
+                    clientRep.setEnabled(false);
+                });
+                fail();
+            } catch (ClientPolicyException cpe) {
+                assertEquals(ClientPolicyEvent.UPDATED.toString(), cpe.getError());
+            }
+            assertEquals(false, getClientByAdmin(clientId).isEnabled());
+
+            updateExecutor(executorName, (ComponentRepresentation provider) -> {
+                provider.getConfig().put(TestRaiseExeptionExecutorFactory.TARGET_CP_EVENTS, new ArrayList<>(Arrays.asList(ClientPolicyEvent.VIEW.toString())));
```

Comment:
Do we need to convert to mutable list here?
`new ArrayList<>(Arrays.asList(ClientPolicyEvent.VIEW.toString()))`
could be simply
`Arrays.asList(ClientPolicyEvent.VIEW.toString())`